### PR TITLE
Fix Pylint warnings for the new checkers

### PR DESCRIPTION
**Pull request by ostoliarova-msft**
# Description

- Renamed enums to be uppercase
- Removed disabling of lint rules for addressed items
- Added a keyword-only api_version argument:  ":keyword api_version:" to client constructors

For more information, please check out the [issue ](https://github.com/Azure/azure-sdk-for-python/issues/24146)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by elavarasidc** [comment]
The changes, look fine to me.
Adding in @shirleyqin-msft to pull in the changes locally, validate and sign off for Rooms.

cc: @radubulboaca

**Review by shirleyqin-msft — APPROVED** [review]

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py [hunk 1]
```diff
@@ -56,7 +61,7 @@ class ChatClient(object): # pylint: disable=client-accepts-api-version-keyword
     def __init__(
             self,
             endpoint,  # type: str
-            credential,  # type: CommunicationTokenCredential
+            credential: Union[TokenCredential, AzureKeyCredential],
```
Comment (ostoliarova-msft) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:64]: Chat team, is this change fine here? Method was regenerated with a new signature

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py [hunk 2]
```diff
@@ -24,24 +24,29 @@
...
     get chat thread client by thread id, list chat threads.
 
     :param str endpoint:
         The endpoint of the Azure Communication resource.
-    :param CommunicationTokenCredential credential:
-        The credentials with which to authenticate.
+    :param Union[TokenCredential, AzureKeyCredential] credential:
+        The credential we use to authenticate against the service.
+
+    :keyword api_version: Api Version. Default value is "2021-09-07". Note that overriding this
```
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:47]:
```suggestion
    :keyword api_version: Azure Communication Chat API Version. Default value is "2021-09-07". Note that overriding this
```

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py [hunk 3]
```diff
@@ -54,12 +55,15 @@ class ChatThreadClient(object): # pylint: disable=client-accepts-api-version-key
...
         The endpoint of the Azure Communication resource.
-    :param CommunicationTokenCredential credential:
-        The credentials with which to authenticate. The value contains a User
-        Access Token
+    :param Union[TokenCredential, AzureKeyCredential] credential:
+        The credential we use to authenticate against the service.
     :param str thread_id:
         The unique thread id.
 
+    :keyword api_version: Api Version. Default value is "2021-09-07". Note that overriding this
```
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py:63]:
```suggestion
    :keyword api_version: Azure Communication Chat API Version. Default value is "2021-09-07". Note that overriding this
```

File: sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_client_async.py [hunk 4]
```diff
@@ -35,18 +36,22 @@
...
     get chat thread client by thread id, list chat threads.
 
     :param str endpoint:
         The endpoint of the Azure Communication resource.
-    :param CommunicationTokenCredential credential:
-        The credentials with which to authenticate.
+    :param Union[TokenCredential, AzureKeyCredential] credential:
+        The credential we use to authenticate against the service.
+
+    :keyword api_version: Api Version. Default value is "2021-09-07". Note that overriding this
```
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_client_async.py:52]:
```suggestion
    :keyword api_version: Azure Communication Chat API. Default value is "2021-09-07". Note that overriding this
```

File: sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_thread_client_async.py [hunk 5]
```diff
@@ -53,12 +55,15 @@ class ChatThreadClient(object): # pylint: disable=client-accepts-api-version-key
...
         The endpoint of the Azure Communication resource.
-    :param CommunicationTokenCredential credential:
-        The credentials with which to authenticate. The value contains a User
-        Access Token
+    :param Union[TokenCredential, AzureKeyCredential] credential:
+        The credential we use to authenticate against the service.
     :param str thread_id:
         The unique thread id.
 
+    :keyword api_version: Api Version. Default value is "2021-09-07". Note that overriding this
```
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_thread_client_async.py:63]:
```suggestion
    :keyword api_version: Azure Communication Chat API. Default value is "2021-09-07". Note that overriding this
```

File: sdk/communication/azure-communication-chat/swagger/SWAGGER.md [hunk 6]
```diff
@@ -17,7 +17,7 @@ autorest SWAGGER.md
 
 ```yaml
 tag: package-chat-2021-09-07
-require: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/communication/data-plane/Chat/readme.md
+require: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/8dbeba81f3a838cd4b7efd70234f29cc1cdc7374/specification/communication/data-plane/Chat/readme.md
```
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/swagger/SWAGGER.md:20]: why did we change the commit hash?

File: sdk/communication/azure-communication-email/azure/communication/email/_email_client.py [hunk 7]
```diff
@@ -24,6 +26,9 @@ class EmailClient(object): # pylint: disable=client-accepts-api-version-keyword
         The endpoint url for Azure Communication Service resource.
     :param Union[TokenCredential, AzureKeyCredential] credential:
         The credential we use to authenticate against the service.
+    :keyword api_version: Api Version. Default value is "2021-10-01-preview". Note that overriding
```
Comment (AikoBB) [inline sdk/communication/azure-communication-email/azure/communication/email/_email_client.py:29]:
```suggestion
    :keyword api_version: Azure Communication Email API . Default value is "2021-10-01-preview". Note that overriding
```

File: sdk/communication/azure-communication-email/azure/communication/email/aio/_email_client_async.py [hunk 8]
```diff
@@ -24,6 +25,9 @@ class EmailClient(object): # pylint: disable=client-accepts-api-version-keyword
         The endpoint url for Azure Communication Service resource.
     :param Union[AsyncTokenCredential, AzureKeyCredential] credential:
         The credential we use to authenticate against the service.
+    :keyword api_version: Api Version. Default value is "2021-10-01-preview". Note that overriding
```
Comment (AikoBB) [inline sdk/communication/azure-communication-email/azure/communication/email/aio/_email_client_async.py:28]:
```suggestion
    :keyword api_version: Azure Communication Email API . Default value is "2021-10-01-preview". Note that overriding
```

File: sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py [hunk 9]
```diff
@@ -21,13 +21,16 @@
...
 
-class CommunicationIdentityClient(object): # pylint: disable=client-accepts-api-version-keyword
+class CommunicationIdentityClient(object):
     """Azure Communication Services Identity client.
 
     :param str endpoint:
         The endpoint url for Azure Communication Service resource.
     :param TokenCredential credential:
         The TokenCredential we use to authenticate against the service.
+    :keyword api_version: Api Version. Default value is "2022-06-01". Note that overriding this
```
Comment (AikoBB) [inline sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py:31]:
```suggestion
    :keyword api_version: Azure Communication Identity API version. Default value is "2022-06-01". Note that overriding this
```

File: sdk/communication/azure-communication-identity/azure/communication/identity/aio/_communication_identity_client_async.py [hunk 10]
```diff
@@ -20,13 +20,16 @@
...
 
-class CommunicationIdentityClient: # pylint: disable=client-accepts-api-version-keyword
+class CommunicationIdentityClient:
     """Azure Communication Services Identity client.
 
     :param str endpoint:
         The endpoint url for Azure Communication Service resource.
     :param AsyncTokenCredential credential:
         The AsyncTokenCredential we use to authenticate against the service.
+    :keyword api_version: Api Version. Default value is "2022-06-01". Note that overriding this
```
Comment (AikoBB) [inline sdk/communication/azure-communication-identity/azure/communication/identity/aio/_communication_identity_client_async.py:30]:
```suggestion
    :keyword api_version: Azure Communication Identity API version. Default value is "2022-06-01". Note that overriding this
```

File: sdk/communication/azure-communication-networktraversal/azure/communication/networktraversal/_communication_relay_client.py [hunk 11]
```diff
@@ -14,20 +14,23 @@
...
-class CommunicationRelayClient(object): # pylint: disable=client-accepts-api-version-keyword
+class CommunicationRelayClient(object):
     """Azure Communication Services Relay client.
 
     :param str endpoint:
         The endpoint url for Azure Communication Service resource.
     :param TokenCredential credential:
         The TokenCredential we use to authenticate against the service.
-
+    :keyword api_version: Api Version. Default value is "2022-03-01-preview". Note that overriding
```
Comment (AikoBB) [inline sdk/communication/azure-communication-networktraversal/azure/communication/networktraversal/_communication_relay_client.py:31]:
```suggestion
    :keyword api_version: Azure Communication Network Traversal API version. Default value is "2022-03-01-preview". Note that overriding
```

File: sdk/communication/azure-communication-networktraversal/azure/communication/networktraversal/aio/_communication_relay_client_async.py [hunk 12]
```diff
@@ -19,13 +20,16 @@
...
 
-class CommunicationRelayClient: # pylint: disable=client-accepts-api-version-keyword
+class CommunicationRelayClient:
     """Azure Communication Services Network Traversal client.
 
     :param str endpoint:
         The endpoint url for Azure Communication Service resource.
     :param AsyncTokenCredential credential:
         The AsyncTokenCredential we use to authenticate against the service.
+    :keyword api_version: Api Version. Default value is "2022-03-01-preview". Note that overriding
```
Comment (AikoBB) [inline sdk/communication/azure-communication-networktraversal/azure/communication/networktraversal/aio/_communication_relay_client_async.py:30]:
```suggestion
    :keyword api_version: Azure Communication Network Traversal API version. Default value is "2022-03-01-preview". Note that overriding
```

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py [hunk 13]
```diff
@@ -20,14 +21,17 @@
...
-class PhoneNumbersClient(object): # pylint: disable=client-accepts-api-version-keyword
+class PhoneNumbersClient(object):
     """A client to interact with the AzureCommunicationService Phone Numbers gateway.
 
     This client provides operations to interact with the phone numbers service
     :param str endpoint:
         The endpoint url for Azure Communication Service resource.
     :param TokenCredential credential:
         The credentials with which to authenticate.
+    :keyword api_version: Api Version. The default value is "2022-01-11-preview2". Note that
```
Comment (AikoBB) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:32]:
```suggestion
    :keyword api_version: Azure Communication Phone Number API version. The default value is "2022-01-11-preview2". Note that
```

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/aio/_phone_numbers_client_async.py [hunk 14]
```diff
@@ -21,14 +22,17 @@
...
-class PhoneNumbersClient(object): # pylint: disable=client-accepts-api-version-keyword
+class PhoneNumbersClient(object):
     """A client to interact with the AzureCommunicationService Phone Numbers gateway.
 
     This client provides operations to interact with the phone numbers service
     :param str endpoint:
         The endpoint url for Azure Communication Service resource.
     :param AsyncTokenCredential credential:
         The credentials with which to authenticate.
+    :keyword api_version: Api Version. The default value is "2022-01-11-preview2". Note that
```
Comment (AikoBB) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/aio/_phone_numbers_client_async.py:33]:
```suggestion
    :keyword api_version: Azure Communication Phone Number API version. The default value is "2022-01-11-preview2". Note that
```

File: sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py [hunk 15]
```diff
@@ -38,6 +38,9 @@ class RoomsClient(object): # pylint: disable=client-accepts-api-version-keyword
         The endpoint url for Azure Communication Service resource.
     :param ~azure.core.credentials.AzureKeyCredential credential:
         The access key we use to authenticate against the service.
+    :keyword api_version: Api Version. Default value is "2022-02-01". Note that overriding this
```
Comment (AikoBB) [inline sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py:41]:
```suggestion
    :keyword api_version: Azure Communication Rooms API version. Default value is "2022-02-01". Note that overriding this
```

File: sdk/communication/azure-communication-rooms/azure/communication/rooms/aio/_rooms_client_async.py [hunk 16]
```diff
@@ -34,6 +34,9 @@ class RoomsClient(object): # pylint: disable=client-accepts-api-version-keyword
         The endpoint url for Azure Communication Service resource.
     :param ~azure.core.credentials.AzureKeyCredential credential:
         The access key we use to authenticate against the service.
+    :keyword api_version: Api Version. Default value is "2022-02-01". Note that overriding this
```
Comment (AikoBB) [inline sdk/communication/azure-communication-rooms/azure/communication/rooms/aio/_rooms_client_async.py:37]:
```suggestion
    :keyword api_version: Azure Communication Rooms API version. Default value is "2022-02-01". Note that overriding this
```

File: sdk/communication/azure-communication-sms/azure/communication/sms/_sms_client.py [hunk 17]
```diff
@@ -26,6 +27,9 @@ class SmsClient(object): # pylint: disable=client-accepts-api-version-keyword
         The endpoint url for Azure Communication Service resource.
     :param TokenCredential credential:
         The TokenCredential we use to authenticate against the service.
+    :keyword api_version: Api Version. The default value is "2021-03-07". Note that
```
Comment (AikoBB) [inline sdk/communication/azure-communication-sms/azure/communication/sms/_sms_client.py:30]:
```suggestion
    :keyword api_version: Azure Communication SMS API version. The default value is "2021-03-07". Note that
```

File: sdk/communication/azure-communication-sms/azure/communication/sms/aio/_sms_client_async.py [hunk 18]
```diff
@@ -26,6 +27,9 @@ class SmsClient(object): # pylint: disable=client-accepts-api-version-keyword
         The endpoint url for Azure Communication Service resource.
     :param AsyncTokenCredential credential:
         The AsyncTokenCredential we use to authenticate against the service.
+    :keyword api_version: Api Version. The default value is "2021-03-07". Note that
```
Comment (AikoBB) [inline sdk/communication/azure-communication-sms/azure/communication/sms/aio/_sms_client_async.py:30]:
```suggestion
    :keyword api_version: Azure Communication SMS API version. The default value is "2021-03-07". Note that
```

**Comment by AikoBB** [comment]
@ostoliarova-msft pls apply the pylint changes for the [azure-communication-jobrouter](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/communication/azure-communication-jobrouter) as well

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py (hunk 1 above)
Comment (AikoBB) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:64]: @LuChen-Microsoft need your confirmation here

**Comment by AikoBB** [comment]
> **API change check**
> 
> APIView has identified API level changes in this PR and created following API reviews.
> 
> [azure-communication-identity](https://apiview.dev/Assemblies/Review/37a7defbbc284d3cb58f78f5e3b4fdbb) [azure-communication-phonenumbers](https://apiview.dev/Assemblies/Review/6bf7ca79a0f148248ac1107fbcf4ca46) [azure-communication-networktraversal](https://apiview.dev/Assemblies/Review/9c2445b7579a4b9e9390b19944bec371) [azure-communication-rooms](https://apiview.dev/Assemblies/Review/a3a22b8c0f994bfa87023ca6668e4227) [azure-communication-chat](https://apiview.dev/Assemblies/Review/6af9f4d070624f8eb817c799082f3b3c) [azure-communication-sms](https://apiview.dev/Assemblies/Review/e058cc51fec24e428cfd4e1fba717017)

@annatisch this PR includes implementation of [Fix Pylint warnings for the new checkers ](https://github.com/Azchaure/azure-sdk-for-python/issues/24146) issue and we are introducing the following API changes, I was wondering if these changes should be released as a minor version release

File: sdk/communication/azure-communication-chat/swagger/SWAGGER.md (hunk 6 above)
Comment (ostoliarova-msft) [reply sdk/communication/azure-communication-chat/swagger/SWAGGER.md:20]: @AikoBB I made it in correspondence with other modalities where swagger is generated based on the specific commit and not on the branch.  Do I need to rollback this change and regenerate from swagger in `main`?

File: sdk/communication/azure-communication-chat/azure/communication/chat/_api_versions.py [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+from enum import Enum
+from azure.core import CaseInsensitiveEnumMeta
+
+
+class ApiVersion(str, Enum, metaclass=CaseInsensitiveEnumMeta):
+    V2021_09_07 = "2021-09-07"
```
[the remark this answers is not in the document]
Comment (ostoliarova-msft) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_api_versions.py:11]: [_azure_communication_chat_service](https://github.com/ostoliarova-msft/azure-sdk-for-python/blob/ostoliarova/sdk-communication/pylint_warnings/sdk/communication/azure-communication-chat/azure/communication/chat/_generated/_azure_communication_chat_service.py#L33) lists "2021-09-07" as default version. However, there is no such version in the link provided by @AikoBB (there are versions "2021-**03**-07" and "2021-09-**27**"). Could you please advise, what versions should be included in _api_versions.py?

[the remark this answers is not in the document]
Comment (AikoBB) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_api_versions.py:11]: @LuChen-Microsoft

File: sdk/communication/azure-communication-chat/swagger/SWAGGER.md (hunk 6 above)
Comment (AikoBB) [reply sdk/communication/azure-communication-chat/swagger/SWAGGER.md:20]: It's up to the chat team, if they want to go with the main branch then it's better to revert as it was. @LuChen-Microsoft need your approval here as well :)

**Review by danielortega-msft — APPROVED** [review]

**Comment by jiriburant** [comment]
Failing tests for sip routing shouldnt be a blocker, it's a problem with test data that we're going to fix.

**Review by AikoBB — APPROVED** [review]
Thank you so much for fixing it for the whole ACS modalities! Once it's merged into the main branch pls get approval from Anna Tisch regarding the apivew changes

**Review by yogeshmo — APPROVED** [review]
The email changes look good to me - thanks!

**Review by sarkar-rajarshi — APPROVED** [review]
Looks for job router

**Comment by AikoBB** [comment]
@ostoliarova-msft one last thing, pls mention that we are providing the ability specify the API version by `api_version` keyword param in the changelog files under the `Features Added` section for each modality

**Review by petrsvihlik — APPROVED** [review]

**Review by radubulboaca — APPROVED** [review]

**Pull request merged**